chatlas/_tokens.py

```diff
@@ -71,17 +80,76 @@ def tokens_reset() -> None:
     _token_counter = ThreadSafeTokenCounter()
 
 
+class TokenPrice(TypedDict):
+    """
+    Defines the necessary information to look up pricing for a given turn.
+    """
+
+    provider: str
+    model: str
+    cached_input: float
+    input: float
+    output: float
+
+
+# Load in pricing pulled from ellmer
+f = resources.files("chatlas").joinpath("data/prices.json").read_text(encoding="utf-8")
+PricingList: list[TokenPrice] = orjson.loads(f)
+
+
+def get_token_pricing(name: str, model: str) -> TokenPrice | dict:
+    """
+    Get the token pricing for the chat if available based on the prices.json file.
+
+    Returns
+    -------
+    dict[str, str | float]
+        A dictionary with the token pricing for the chat. The keys are:
```

Comment:
The more "standard" way to approach docs here is to keep this part minimal, but move the details to the `TokenPrice` class